draft-ietf-quic-multipath.md

```diff
@@ -438,62 +437,29 @@ connectivity or local preferences. After an endpoint abandons
-
-Either endpoint can initiate path closure
-by sending a PATH_ABANDON frame (see {{path-abandon-frame}}) which
-requests the peer to stop sending packets on the the path with
-the corresponding Path ID.
-
-When a path is abandoned, all connection IDs allocated by both
-of the endpoints for the specified Path ID need to be retired.
-When sending or receiving a PATH_ABANDON frame, endpoints SHOULD wait for at
-least three times the current Probe Timeout (PTO) interval after the last
-packet was sent on the path, as defined in {{Section 6.2 of QUIC-RECOVERY}},
-before sending MP_RETIRE_CONNECTION_ID frames.
-This is inline with the requirement of {{Section 10.2 of QUIC-TRANSPORT}}.
-Both endpoints SHOULD send MP_RETIRE_CONNECTION_ID frames
-for all connection IDs associated to the Path ID of the abandoned path
-to ensure that paths close cleanly and that delayed or reordered packets
-are properly discarded.
-The effect of receiving a MP_RETIRE_CONNECTION_ID frame is specified in
-Section {{retire-cid-close}}.
-
-Usually, it is expected that the PATH_ABANDON frame is used by the client
-to indicate to the server that path conditions have changed such that
-the path is or will be not usable anymore, e.g. in case of a mobility
-event. The PATH_ABANDON frame therefore recommends to the receiver
-that no packets should be sent on that path anymore.
-In addition, the MP_RETIRE_CONNECTION_ID frame is used indicate to the receiving peer
-that the sender will not send any packets associated to the
-connection ID used on that path anymore.
-The receiver of a PATH_ABANDON frame MAY also send
-a PATH_ABANDON frame to indicate its own unwillingness to receive
-any packet on this path anymore.
-
-Reception or sending of the PATH_ABANDON frame is
-the first step to release all resources related to a
-Path ID. However, the Path ID can only be released after all active
-connection IDs for the Path ID have been retired or timed-out after
-the PATH_ABANDON frame was sent.
-Still, when an endpoint receives an PATH_ABANDON frame,
-it SHOULD NOT use the associated Path ID in future frames, except
-in ACK_MP frames for acknowledging inflight packets and
-in MP_RETIRE_CONNECTION_ID frames for connection ID retirement.
+This is true whether the decision to close the path results
+from implicit signals such as idle time or packet losses
+(see {(idle-time-close}}) or for any other reason, such as management
+of local resources. It is also possible to abandon a path for which no
+packet has been sent (see {{abandon-early}}).
+
+When an endpoint receives a PATH_ABANDON frame, it MUST send a corresponding
+PATH_ABANDON frame if it has not already done so, and it MUST treat all
+connection identifiers received from the peer for that path as immediately
+retired. However, the endpoint should retain some knowledge of the
+connection identifiers sent to the peer for a short period,
+as specified in {{spurious-stateless-reset}}.
```

Comment:
I tried to develop the ACK_MP issue in the subsection "Handling ACK_MP for abandoned paths {#ack-after-abandon}". Maybe we should call that out here, in the same way that we call out stateless reset.

Reply:
I also think that we have a hard rule on the data sender side: "if a path info is deleted, then all pending packets MUST be considered lost", and softer rule "don't delete the info too soon because you will end up with spurious retransmit". On the data receiver side, we could have a hard rule, "you MUST send an ACK_MP for all packets received before abandon by the peer", and a softer rule "you should wait a bit for packets to arrive after abandon so you can ACK them and avoid forcing the peer to resend them". Maybe develop that a bit more in the dedicated subsection.

Reply:
@mirjak I think that the "tighten" commit fixes this. Please review.